client/src/hooks/datafiles/mutations/useCompress.ts

```diff
@@ -0,0 +1,156 @@
+import { useMutation } from '@tanstack/react-query';
+import { useSelector, useDispatch, shallowEqual } from 'react-redux';
+import { getCompressParams } from 'utils/getCompressParams';
+import { apiClient } from 'utils/apiClient';
+import { TTapisFile, TPortalSystem } from 'utils/types';
+import { TJobBody, TJobPostResponse } from './useSubmitJob';
+
+async function submitJobUtil(body: TJobBody) {
+  const res = await apiClient.post<TJobPostResponse>(
+    `/api/workspace/jobs`,
+    body
+  );
+  return res.data.response;
+}
+
+function useCompress() {
+  const dispatch = useDispatch();
+  const status = useSelector(
+    (state: any) => state.files.operationStatus.compress,
+    shallowEqual
+  );
+
+  const setStatus = (newStatus: any) => {
+    dispatch({
+      type: 'DATA_FILES_SET_OPERATION_STATUS',
+      payload: { status: newStatus, operation: 'compress' },
+    });
+  };
+
+  const compressErrorAction = (errorMessage: any) => {
+    return {
+      type: 'DATA_FILES_SET_OPERATION_STATUS',
+      payload: {
+        status: { type: 'ERROR', message: errorMessage },
+        operation: 'compress',
+      },
+    };
+  };
+
+  const compressApp = useSelector(
+    (state: any) => state.workbench.config.compressApp
+  );
+
+  const defaultAllocation = useSelector(
+    (state: any) =>
+      state.allocations.portal_alloc || state.allocations.active[0].projectName
+  );
+
+  const systems = useSelector(
+    (state: any) => state.systems.storage.configuration
+  );
+
+  const { mutateAsync } = useMutation({ mutationFn: submitJobUtil });
+
+  const compress = ({
+    scheme,
+    files,
+    filename,
+    compressionType,
+  }: {
+    scheme: string;
+    files: TTapisFile[];
+    filename: string;
+    compressionType: string;
+  }) => {
+    dispatch({
+      type: 'DATA_FILES_SET_OPERATION_STATUS',
+      payload: { status: 'RUNNING', operation: 'compress' },
+    });
+
+    let defaultPrivateSystem: TPortalSystem | undefined;
+
+    if (files[0].scheme === 'private' && files[0].api === 'tapis') {
+      defaultPrivateSystem === null;
```

Comment:
This is not assignment, it should be `defaultPrivateSystem = null`

Reply:
Gotcha. I'll fix that.

Reply:
Trying to put `defaultPrivateSystem = null` is giving errors. I'll see if I can fix this.